app/src/main/java/org/eclipse/kuksa/testapp/databroker/JavaDataBrokerEngine.java

```diff
@@ -33,11 +33,13 @@
 import org.eclipse.kuksa.connectivity.databroker.request.UpdateRequest;
```

Comment:
Functional Finding: 

when trying to set Array Values in VSS Paths Mode (e.g. Vehicle.Cabin.Infotainment.SmartphoneProjection.SupportedMode) we have a crash:

```
FATAL EXCEPTION: main
Process: org.eclipse.kuksa.testapp, PID: 32742
java.lang.UnsupportedOperationException
    at com.google.protobuf.AbstractProtobufList.ensureIsMutable(AbstractProtobufList.java:177)
    at com.google.protobuf.AbstractProtobufList.addAll(AbstractProtobufList.java:110)
    at org.eclipse.kuksa.extension.DataPointExtensionKt.createStringArray(DataPointExtension.kt:234)
    at org.eclipse.kuksa.extension.DataPointExtensionKt.createDatapoint(DataPointExtension.kt:116)
    at org.eclipse.kuksa.testapp.databroker.vsspaths.viewmodel.VSSPathsViewModel.getDatapoint(VSSPathsViewModel.kt:64)
    at org.eclipse.kuksa.testapp.databroker.vsspaths.view.VssPathsViewKt$VssPathsView$1$4$2.invoke(VssPathsView.kt:184)
    at org.eclipse.kuksa.testapp.databroker.vsspaths.view.VssPathsViewKt$VssPathsView$1$4$2.invoke(VssPathsView.kt:181)
...
```

Might not be in scope of your ticket, but maybe it's a simple fix to simply use 
```
// createStringArray
csvValues.forEach {
    array.valuesList.add(it)
}
```
It would be cool if you could try to fix it, but if it's more effort than the above suggestion, I can create a ticket instead.

Reply:
Works for string arrays now but IntArray etc. support needs to be fixed with: #109 

You will still get a type mismatch for these:

<img width="301" alt="Screenshot 2024-03-25 at 12 40 15" src="https://github.com/eclipse-kuksa/kuksa-android-sdk/assets/1355532/ab822916-3943-40da-8fe3-7f19e102a9ff">